candidates/tests/auth.py

```diff
@@ -23,11 +23,14 @@ def setUpClass(cls):
                 ('ermintrude', 'user_who_can_rename', [TRUSTED_TO_RENAME_GROUP_NAME]),
                 ('frankie', 'user_who_can_record_results', [RESULT_RECORDERS_GROUP_NAME]),
         ):
-            u = User.objects.create_user(
-                username,
-                username + '@example.com',
-                'notagoodpassword',
-            )
+            try:
+                u = User.objects.get(username=username)
+            except User.DoesNotExist:
```

Comment:
Yes, I stupidly was on the wrong branch for some of my attempts to reproduce it - I've managed that now. One way this can happen with the code as it is currently is if an exception is thrown in `setUpClass` - in that situation `tearDownClass` isn't run, so the `User` objects will persist.

The reason that didn't quite make sense to me, though, is that since Django 1.8 (and in particular this commit https://github.com/django/django/commit/da9fe5c717c179a9e881a4 ) `setUpClass` and `tearDownClass` should have been in a transaction, so even if `tearDownClass` wasn't run, it wouldn't matter - the rollback of the transaction would have removed the `User` objects.

However, I think we're not getting that class-wide transaction isolation because these two methods don't call the class methods they're overriding, as the [warning here ](https://docs.djangoproject.com/en/1.8/topics/testing/tools/#django.test.SimpleTestCase) says to. If I apply this patch:

```diff
diff --git a/candidates/tests/auth.py b/candidates/tests/auth.py
index de549bc..9a19892 100644
--- a/candidates/tests/auth.py
+++ b/candidates/tests/auth.py
@@ -14,6 +14,7 @@ class TestUserMixin(object):
 
     @classmethod
     def setUpClass(cls):
+        super(TestUserMixin, cls).setUpClass()
         cls.users_to_delete = []
         for username, attr, group_names in (
                 ('john', 'user', []),
@@ -47,5 +48,6 @@ class TestUserMixin(object):
 
     @classmethod
     def tearDownClass(cls):
+        super(TestUserMixin, cls).tearDownClass()
         for u in cls.users_to_delete:
             u.delete()
```

... then there shouldn't be any User objects left behind in any situation (and that does seem to be the case in my cursory testing.) So I think that would be a better patch overall.